src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -250,14 +262,31 @@ public static boolean onLivingUseTotem(LivingEntity entity, DamageSource damageS
         return !NeoForge.EVENT_BUS.post(new LivingUseTotemEvent(entity, damageSource, totem, hand)).isCanceled();
     }
 
-    public static float onLivingHurt(LivingEntity entity, DamageSource src, float amount) {
-        LivingHurtEvent event = new LivingHurtEvent(entity, src, amount);
-        return (NeoForge.EVENT_BUS.post(event).isCanceled() ? 0 : event.getAmount());
+    public static float onIncomingDamage(LivingEntity entity, DamageContainer container) {
+        return NeoForge.EVENT_BUS.post(new IncomingDamageEvent(entity, container)).getDamageContainer().getNewDamage();
+    }
+
+    public static void onLivingDamageTaken(LivingEntity entity, DamageContainer container) {
+        NeoForge.EVENT_BUS.post(new DamageTakenEvent(entity, new DamageContainer.ResultDamageContainer(container)));
     }
 
-    public static float onLivingDamage(LivingEntity entity, DamageSource src, float amount) {
-        LivingDamageEvent event = new LivingDamageEvent(entity, src, amount);
-        return (NeoForge.EVENT_BUS.post(event).isCanceled() ? 0 : event.getAmount());
+    public static void onArmorHurt(DamageSource source, NonNullList<ItemStack> armor, float damage, Player player) {
+        EnumMap<EquipmentSlot, ArmorHurtEvent.ArmorEntry> armorMap = new EnumMap<>(EquipmentSlot.class);
+        for (int index : Inventory.ALL_ARMOR_SLOTS) {
+            ItemStack armorPiece = armor.get(index);
+            if (armorPiece.isEmpty()) continue;
+            EquipmentSlot slot = EquipmentSlot.byTypeAndIndex(EquipmentSlot.Type.ARMOR, index);
+            armorMap.put(slot, new ArmorHurtEvent.ArmorEntry(armorPiece, damage));
+        }
+
+        ArmorHurtEvent event = NeoForge.EVENT_BUS.post(new ArmorHurtEvent(armorMap, player));
+        if (event.isCanceled()) return;
+        event.getArmorMap().forEach((slot, entry) -> {
+            if ((!source.is(DamageTypeTags.IS_FIRE) || !entry.armorItemStack.getItem().isFireResistant()) && entry.armorItemStack.getItem() instanceof ArmorItem) {
```

Comment:
shouldn't this be done in the loop before the event instead of after? For non-armor items, either don't include it in `armorMap` at all, or include it with 0 as long as its not empty (so people can damage non-armor items).

For fire damage, just set it to 0 if the armor stack is fire resistance and its fire damage, but let the event override that behavior if they wish.

Reply:
I like that change, yeah.  i'll add that now.